libdnf5/rpm/solv/goal_private.cpp

```diff
@@ -270,6 +284,7 @@ bool GoalPrivate::limit_installonly_packages(libdnf5::solv::IdQueue & job, Id ru
 
         const InstallonlyCmpData installonly_cmp_data{spool, running_kernel};
         q.sort(&installonly_cmp, &installonly_cmp_data);
+        std::sort(available_unused_providers.begin(), available_unused_providers.end(), nevra_solvable_cmp_key);
```

Comment:
If I am not mistaken, the original patch for libdnf uses name cmp key and not nevra for both - sorting and lower_bound. May I ask you why is there a different implementation?

Reply:
Oh right, I forgot about that.
I think I added that just because `nevra_solvable_cmp_key` is already implemented.
I can add a new function to compare just the names if you prefer that?
The `arch` and `evr` sorting is not needed here so it would be more efficient.

Reply:
Yes sorting using only name will be more efficient,  but sorting by NEVRA is not wrong. Please feel free to implement it according to your preferences.

Reply:
Ok, I have added the new comparator for names only.

I have also rebased the PR.